packages/server/dataloader/customLoaderMakers.ts

```diff
@@ -125,31 +132,67 @@ export const userTasks = (parent: RethinkDataLoader) => {
   return new DataLoader<UserTasksKey, Task[], string>(
     async (keys) => {
       const r = await getRethink()
-      const userIds = keys.map(({userId}) => userId)
-      const teamIds = Array.from(new Set(keys.flatMap(({teamIds}) => teamIds)))
-      const taskLoader = parent.get('tasks')
-      const allUsersTasks = (await r
-        .table('Task')
-        .getAll(r.args(userIds), {index: 'userId'})
-        .filter((task) => r(teamIds).contains(task('teamId')))
-        .filter((task) =>
-          task('tags')
-            .contains('archived')
-            .not()
+      const uniqKeys = keys.reduce((dedupedKeys, currKey) => {
```

Comment:
nit: probably a premature optimization, but you could reduce the complexity of this by using a set.
```ts
const uniqKeys = [] as UserTasksKey[]
const keySet = new Set()
keys.forEach((key, idx) => {
  const serialized = serializeUserTaskKey(key)
  if (!keySet.has(serialized)) {
    keySet.add(serialized);
    uniqKeys.push(key)
  }
}
```